.github/workflows/build.yml

```
@@ -14,49 +28,107 @@
+            echo "Build required: Relevant files changed"
+          # Skip build
+          else
+            echo "reason=skip" >> $GITHUB_OUTPUT
+            echo "should_run=false" >> $GITHUB_OUTPUT
+            echo "Build skipped: No relevant changes and no release"
+          fi
+
       # Setup Golang
       - name: 🐿 Setup Golang
+        if: steps.should-build.outputs.should_run == 'true'
         uses: actions/setup-go@v4
```

actionlint (1.7.7)

the runner of "actions/setup-go@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)